packages/perseus/src/widgets/interactive-graphs/graphs/segment.tsx

```diff
@@ -1,121 +1,79 @@
 import type {MafsGraphProps} from "../types";
 
-export type SegmentProps = MafsGraphProps<PerseusGraphTypeSegment>;
-
-const getSegmentCoords = (
-    props: SegmentProps,
-): ReadonlyArray<ReadonlyArray<Coord>> => {
-    const coords = props.graph.coords;
-    if (coords) {
-        return coords;
-    }
-
-    const ys = (n?: number) => {
-        switch (n) {
-            case 2:
-                return [5, -5];
-            case 3:
-                return [5, 0, -5];
-            case 4:
-                return [6, 2, -2, -6];
-            case 5:
-                return [6, 3, 0, -3, -6];
-            case 6:
-                return [5, 3, 1, -1, -3, -5];
-            default:
-                return [5];
-        }
-    };
-
-    return ys(props.graph.numSegments).map((y) => {
-        let segment: Coord[] = [
-            [-5, y],
-            [5, y],
-        ];
-        segment = normalizeCoords(segment, props.range);
-        segment = normalizePoints(props.range, props.step, segment);
-        return segment;
-    });
-};
-
-const updateSegmentsArray = (
-    segments: ReadonlyArray<ReadonlyArray<Coord>>,
-    index: number,
-    segment: ReadonlyArray<Coord>,
-) => segments.map((seg: any, i: any) => (index === i ? segment : seg));
+type SegmentProps = MafsGraphProps<SegmentGraphState>;
 
 export const SegmentGraph = (props: SegmentProps) => {
-    const segments = getSegmentCoords(props);
-
-    const handleChange = (i: number, segment: any) =>
-        props.onGraphChange((current) => ({
-            ...current,
-            coords: current.coords
-                ? updateSegmentsArray(current.coords, i, segment)
-                : updateSegmentsArray(segments, i, segment),
-        }));
+    const {dispatch} = props;
+    const {segments, snapStep, range} = props.graphState;
 
     return (
         <>
             {segments.map((segment, i) => (
-                <Segment
+                <SegmentView
                     key={i}
-                    i={i}
                     segment={segment}
-                    snaps={props.snapStep}
-                    range={props.range}
-                    onChange={handleChange}
+                    snaps={snapStep}
+                    range={range}
+                    onMoveSegment={(delta: vec.Vector2) => {
+                        dispatch(moveSegment(i, delta));
+                    }}
+                    onMovePoint={(
+                        endpointIndex: number,
+                        destination: vec.Vector2,
+                    ) =>
+                        dispatch(
+                            moveControlPoint(i, endpointIndex, destination),
+                        )
+                    }
                     data-testid={"segment" + i}
                 />
             ))}
         </>
     );
 };
 
-const Segment = (props: {
-    i: number;
-    segment: ReadonlyArray<Coord>;
+const SegmentView = (props: {
```

Comment:
It was clashing with the type import from `interactive-graph-state.ts`. As far as TS is concerned, types and functions occupy different namespaces, so the names should not conflict, but I had great difficulty persuading eslint that this was okay. Since this name (`SegmentView`) is local to the file, I figured it would be less damaging to rename it rather than rename the state type to something like `SegmentModel`.